### PR TITLE
Fix lookup embedder normalize

### PR DESCRIPTION
**Pull request by samuelbroscheit**
See discussion here https://github.com/uma-pi1/kge/pull/116

### CODE REVIEW

**Comment by samuelbroscheit** [comment]
Don't review yet.

**Comment by samuelbroscheit** [comment]
Ok, now please review.

File: kge/model/embedder/lookup_embedder.py [hunk 1]
```diff
@@ -61,21 +61,16 @@ def __init__(
...
+                with torch.no_grad():
+                    self._embeddings.weight.data =\
+                        torch.nn.functional.normalize(
+                            self._embeddings.weight.data, p=self.normalize_p, dim=-1
                         )
-
+            # normalize before starting a batch
             job.pre_batch_hooks.append(normalize_embeddings)
+            # normalize after an epoch (before validation and checkpoint is saved)
+            job.post_epoch_hooks.append(normalize_embeddings)
```
Comment (rgemulla) [inline kge/model/embedder/lookup_embedder.py:73]: More suitable would be post_batch_hook. Otherwise looks good.

Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.py:73]: You mean pre_batch_hooks + post_batch_hooks ?

**Comment by rufex2001** [comment]
I'm guessing just normalizing after the updates, so post batch only?

File: kge/model/embedder/lookup_embedder.py (hunk 1 above)
Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.py:73]: @rufex2001 commented 
> I'm guessing just normalizing after the updates, so post batch only?

This would mean that the computation for the first batch after resuming or initializaton would be wrong. Or what do you think?

**Comment by rufex2001** [comment]
True, those forward passes would be wrong...

File: kge/model/embedder/lookup_embedder.py (hunk 1 above)
Comment (rgemulla) [reply kge/model/embedder/lookup_embedder.py:73]: One should normalize anyway during __init__. This handles initialization and loading: the model starts with correct embeddings. Then they are renormalized whenever they are changed, i.e., after each batch.

Comment (samuelbroscheit) [reply kge/model/embedder/lookup_embedder.py:73]: See https://github.com/uma-pi1/kge/pull/124 to enable this change.

**Comment by rgemulla** [comment]
Superseded by #138
